main.tf

```diff
@@ -119,6 +119,42 @@ data "aws_iam_policy" "aws_config_built_in_role" {
   arn = "arn:aws:iam::aws:policy/service-role/AWS_ConfigRole"
 }
 
+#-----------------------------------------------------------------------------------------------------------------------
+# Optionally create an IAM Role for an Aggregator to query Organizations data (if we're the central AWS Config account)
+#-----------------------------------------------------------------------------------------------------------------------
+module "iam_role_organization_aggregator" {
+  count   = module.this.enabled && var.aggregate_organization_wide && local.create_iam_role ? 1 : 0
+  source  = "cloudposse/iam-role/aws"
+  version = "0.9.3"
+
+  principals = {
+    "Service" = ["config.amazonaws.com"]
+  }
+
+  policy_document_count = 0
+  policy_description    = "AWS Config IAM policy for Organization Aggregation"
+  role_description      = "AWS Config IAM role for Organization Aggregation"
+
+  name         = "aggregator"
+  use_fullname = true
+
+  attributes = ["config"]
+
+  context = module.this.context
+}
+
+resource "aws_iam_role_policy_attachment" "config_organization_aggregator_policy_attachment" {
+  count = module.this.enabled && var.aggregate_organization_wide && local.create_iam_role ? 1 : 0
+
+  role       = module.iam_role_organization_aggregator[0].name
+  policy_arn = data.aws_iam_policy.aws_config_built_in_role_for_organizations.arn
+}
+
+data "aws_iam_policy" "aws_config_built_in_role_for_organizations" {
+  arn = "arn:aws:iam::aws:policy/service-role/AWSConfigRoleForOrganizations"
```

Comment:
Consider using the [aws_partition](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition) data source to look up the partition rather than hard-coding `aws` here.